### PR TITLE
fix: add ability to detect mobile soundcloud url

### PR DESCRIPTION
**Pull request by takase1121**
Basically just edited some regexes (didn't touch those for a long time, damn!)
I am not sure if the tests are configured correctly. It seems to work though.

### CODE REVIEW

File: dist/jsVideoUrlParser.min.js [hunk 1]
```diff
@@ -1 +1 @@
...
...:"document",CHANNEL:"channel",COLLECTION:"collection",GROUP:"group"}}return C.prototype.parse=function(e,t){var i={};i.list=this.parsePlaylist(t),i.params=t;var r=e.match(/\/(audio|video|document|user\/channel|collection|group)\/(?:[\w-]+-)?(\w+)/);if(r)return i.mediaType=this.parseMediaType(r[1]),i.id=r[2],i},C.prototype.parsePlaylist=function(e){if(e["playlist-id"]){var t=e["playlist-id"];return delete e["playlist-id"],t}},C.prototype.parseMediaType=function(e){switch(e){case"audio":return this.mediaTypes.AUDIO;case"video":return this.mediaTypes.VIDEO;case"document":return this.mediaTypes.DOCUMENT;case"user/channel":return this.mediaTypes.CHANNEL;case"collection":return this.mediaTypes.COLLECTION;case"group":return this.mediaTypes.GROUP}},C.prototype.createLongUrl=function(e,t){if(e.id){var i="https://www.teachertube.com/";return e.list&&(t["playlist-id"]=e.list),e.mediaType===this.mediaTypes.CHANNEL?i+="user/channel/":i+=e.mediaType+"/",i+=e.id,i+=S(t)}},C.prototype.createEmbedUrl=function(e,t){if(e.id){var i="https://www.teachertube.com/embed/";if(e.mediaType===this.mediaTypes.VIDEO||e.mediaType===this.mediaTypes.AUDIO)return i+=e.mediaType+"/"+e.id,i+=S(t)}},o.bind(new C),o});
```
Comment (Zod-) [inline dist/jsVideoUrlParser.min.js]: Could you undo the changes to the dist file here (probably just amend). I update the stuff in /dist in "release builds"

File: lib/provider/soundcloud.js [hunk 2]
```diff
@@ -22,7 +22,7 @@ module.exports = SoundCloud;
 
 SoundCloud.prototype.parseUrl = function(url, result) {
   var match = url.match(
-    /soundcloud\.com\/(?:([\w-]+)\/(sets\/)?)([\w-]+)/i
+    /(?:m\.)?soundcloud\.com\/(?:([\w-]+)\/(sets\/)?)([\w-]+)/i
```
Comment (Zod-) [inline lib/provider/soundcloud.js]: Should this also support creating mobile urls then?

File: dist/jsVideoUrlParser.min.js (hunk 1 above)
Comment (takase1121) [reply dist/jsVideoUrlParser.min.js]: Wrestled with git for a bit. I think now its reverted to what it should be.

File: lib/provider/soundcloud.js (hunk 2 above)
Comment (takase1121) [reply lib/provider/soundcloud.js]: I don't think there is any benifit to it. The `m.soundcloud.com` url is automatically expanded (at least by my observations) so I don't think there is any use for that.

**Comment by takase1121** [comment]
Is there any issue with this PR that keep it open?

**Comment by Zod-** [comment]
> Is there any issue with this PR that keep it open?

The issue was me forgetting about it, sorry.
